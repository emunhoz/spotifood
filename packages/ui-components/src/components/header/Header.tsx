import React from 'react'
import logo from './img/spotifood-logo.svg'
import * as S from './Header.style'

interface IHeader {
  user?: {
    // eslint-disable-next-line camelcase
    display_name: string
    images: [
      {
        url: string
      }
    ]
  }
  signOut?: () => void
}

export const Header = ({ user, signOut }: IHeader) => (
  <S.Header>
    <S.SpotifoodLogo src={logo} alt='Logo' width='124' height='45' />
    {user && (
      <S.User>
        <S.NameWrapper>
          <S.DisplayName>{user.display_name}</S.DisplayName>
          <S.SignOutLink onClick={() => signOut?.()}>Sair</S.SignOutLink>
        </S.NameWrapper>
        <S.Avatar src={user.images[0].url} alt={user.display_name} />
      </S.User>
    )}
  </S.Header>
)
