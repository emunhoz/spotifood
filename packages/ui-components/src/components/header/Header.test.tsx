import React from 'react'
import { render } from '../../utils/test-utils'
import { Header } from './Header'

describe('[Component]: Header', () => {
  test('render the component', () => {
    const { asFragment } = render(
      <Header
        signOut={() => console.log('sair')}
        user={{
          display_name: 'Eder',
          images: [
            {
              url:
                'https://i.scdn.co/image/ab6775700000ee85ffd83d8fd3711add44d62317'
            }
          ]
        }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
