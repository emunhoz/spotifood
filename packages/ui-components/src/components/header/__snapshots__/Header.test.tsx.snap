// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Header render the component 1`] = `
<DocumentFragment>
  <header
    class="sc-bdnxRM hjGVAm"
  >
    <img
      alt="Logo"
      class="sc-gKAaRy"
      height="45"
      src="test-file-stub"
      width="124"
    />
    <div
      class="sc-gtsrHT ljPrYI"
    >
      <div
        class="sc-hKFxyN ikhIOv"
      >
        <span
          class="sc-eCApnc jfAvdG"
        >
          Eder
        </span>
        <span
          class="sc-jSFjdj cjDpPs"
        >
          Sair
        </span>
      </div>
      <img
        alt="Eder"
        class="sc-dlnjwi jXpYGt"
        src="https://i.scdn.co/image/ab6775700000ee85ffd83d8fd3711add44d62317"
      />
    </div>
  </header>
</DocumentFragment>
`;

exports[`[Component]: Header render the component without user info 1`] = `
<DocumentFragment>
  <header
    class="sc-bdnxRM hjGVAm"
  >
    <img
      alt="Logo"
      class="sc-gKAaRy"
      height="45"
      src="test-file-stub"
      width="124"
    />
  </header>
</DocumentFragment>
`;
